controllers/topology/reconcile_state.go

```diff
@@ -78,7 +78,7 @@ func (r *ClusterReconciler) reconcileControlPlane(ctx context.Context, s *scope.
 
 		cpInfraRef, err := contract.ControlPlane().MachineTemplate().InfrastructureRef().Get(s.Desired.ControlPlane.Object)
 		if err != nil {
-			return errors.Wrapf(err, "failed to update the %s object,", s.Desired.ControlPlane.InfrastructureMachineTemplate.GetKind())
+			return errors.Wrapf(err, "failed to update %s", errObjectID(s.Desired.ControlPlane.InfrastructureMachineTemplate))
```

Comment:
```suggestion
			return errors.Wrapf(err, "failed to update %s", KRef{Obj: s.Desired.ControlPlane.InfrastructureMachineTemplate})
```

@fabriziopandini The idea is to switch to KRef, right?
Just asking because previously our format was:
```go
fmt.Sprintf("%s, %s/%s", object.GetObjectKind().GroupVersionKind().GroupVersion(), object.GetObjectKind().GroupVersionKind().Kind, object.GetName()
```
And KRef does not include the group version:
```go
fmt.Sprintf("%s/%s", ref.Obj.GetObjectKind().GroupVersionKind().Kind, ref.Obj.GetName())
```

But I think it's fine to drop the group version as it will make the error message more concise and I think we don't really need the group version to identify the object.

Reply:
Agreed. Let's KRef